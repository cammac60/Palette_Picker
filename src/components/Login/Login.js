import React, { useState } from 'react';
import { withRouter } from 'react-router-dom';
import './Login.scss';

export const handleChange = (event, setError, setUsername, setPassword) => {
  const { id, value } = event.target;
  setError('');
  if (id === 'username') {
    setUsername(value);
  } else {
    setPassword(value);
  }
};

export const Login = props => {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [error, setError] = useState('');

  const handleSubmit = e => {
    e.preventDefault();
    if (validateSubmit()) {
      props.setUser(username);
      props.history.push('/home');
    } else {
      setError('Both fields are required.');
    }
  };

  const validateSubmit = () => {
    return username && password ? true : false
  };

  return (
    <div className="login">
      <h1>Palette Picker</h1>
      <span className="form-label">Please sign in to start pickin' palettes:</span>
      <form className="login-form" onSubmit={event => handleSubmit(event)}>
        <input className="username" id="username" type="text" placeholder="Username..." value={username} onChange={e => handleChange(e, setError, setUsername, setPassword)} maxlength="32"/>
        <input className="password" id="password" type="password" placeholder="Password..." value={password} onChange={e => handleChange(e, setError, setUsername, setPassword)} maxlength="32"/>
        <button type="submit" className="login-submit">Sign In</button>
      </form>
      <span className="login-error">{error}</span>
    </div>
  )
}

export default withRouter(Login);
