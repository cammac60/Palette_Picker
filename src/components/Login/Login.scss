@import '../../variables.scss';

.login {
  background: {
    image: url('../../images/wave1.png');
    position: center;
    repeat: no-repeat;
    size: cover;
  }
  margin: 0;
  min-height: 100%;
  min-width: 100%;
}

h1, .form-label, .login-form, {
  margin: {
    left: auto;
    right: auto;
  }
  width: 51%;
}

h1 {
  font: {
    family: $title-font;
    size: 5rem;
  }
  text-decoration: underline;
}

.form-label {
  display: block;
  font-size: 1.7rem;
  margin-top: 15%;
  text-align: left;
}

.login-form {
  display: flex;
  height: 40px;
  justify-content: space-between;
  margin-top: 5%;
}

.username, .password {
  border: none;
  border-radius: 4px;
  font-size: 1.6rem;
  padding: 5px;
  width: 30%;
}

.login-submit {
  background-color: $accent-color-1;
  border: none;
  border-radius: 45px;
  color: $dark-text-color;
  font-size: 1.4rem;
  width: 20%;
  &:active {
    background-color: $dark-text-color;
    color: $light-text-color;
  }
}

.login-error {
  background-color: rgba(100, 0, 0, 0.5);
  display: block;
  font-size: 1.1rem;
  height: 40px;
  margin: {
    left: 24.5%;
    right: auto;
    top: 3%;
  }
  padding: 5px;
  text-align: left;
  width: 15.3%;
}
