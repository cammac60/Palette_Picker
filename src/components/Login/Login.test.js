import React from 'react';
import ReactDOM from 'react-dom';
import { Login } from './Login.js';
import { render, fireEvent } from '@testing-library/react';
import { shallow } from 'enzyme';

describe('Login', () => {
  let wrapper, mockSetUser, mockError;

  it('Should match snapshot', () => {
    mockSetUser = jest.fn();
    mockError = '';
    wrapper = shallow(<Login
      setUser={mockSetUser}
      error={mockError}
    />);
    expect(wrapper).toMatchSnapshot();
  })

  it('Should render without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(<Login />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('Should load with initial username of an empty string', () => {
    const { container } = render(<Login />);
    const username = container.querySelector('#username');
    expect(username.textContent).toBe('');
  });
  it('Should load with an initial password of an empty string', () => {
    const { container } = render(<Login />);
    const password = container.querySelector('#password');
    expect(password.textContent).toBe('');
  });
  it('Should load with an initial error of an empty string', () => {
    const { container } = render(<Login />);
    const error = container.querySelector('.login-error');
    expect(error.textContent).toBe('');
  });
  it('Should add an error message if the button is clicked with no values in one or more inputs', () => {
    const { container } = render(<Login />);
    const button = container.querySelector('.login-submit');
    const error = container.querySelector('.login-error');
    fireEvent.click(button);
    expect(error.textContent).toBe('Both fields are required.');
  });
});
