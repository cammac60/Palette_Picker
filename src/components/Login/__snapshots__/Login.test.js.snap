// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Should match snapshot 1`] = `
<div
  className="login"
>
  <h1>
    Palette Picker
  </h1>
  <span
    className="form-label"
  >
    Please sign in to start pickin' palettes:
  </span>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <input
      className="username"
      id="username"
      onChange={[Function]}
      placeholder="Username..."
      type="text"
      value=""
    />
    <input
      className="password"
      id="password"
      onChange={[Function]}
      placeholder="Password..."
      type="password"
      value=""
    />
    <button
      className="login-submit"
      type="submit"
    >
      Sign In
    </button>
  </form>
  <span
    className="login-error"
  />
</div>
`;
